### PR TITLE
[Merged by Bors] - feat: port Data.String.Basic

### PR DESCRIPTION
**Pull request by javra**
mathlib3 SHA: d13b3a4a392ea7273dfa4727dbd1892e26cfd518

### CODE REVIEW

**Comment by javra** [comment]
Finished the missing proof now, so it's only some simp lemmas left now

File: Mathlib/Data/String/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,333 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+
+! This file was ported from Lean 3 source module data.string.basic
+! leanprover-community/mathlib commit 8a275d92e9f9f3069871cbdf0ddd54b88c17e144
```
Comment (eric-wieser) [inline Mathlib/Data/String/Basic.lean:7]: This commit doesn't exist: https://github.com/leanprover-community/mathlib/commit/8a275d92e9f9f3069871cbdf0ddd54b88c17e144

**Comment by ChrisHughes24** [comment]
bors merge

File: Mathlib/Data/String/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    apply le_trans
+  lt_iff_le_not_le a b := by
+    simp only [le_iff_toList_le, lt_iff_toList_lt, lt_iff_le_not_le]
+  le_antisymm a b := by
+    simp only [le_iff_toList_le, ← toList_inj]
+    apply le_antisymm
+  le_total a b := by
+    simp only [le_iff_toList_le]
+    apply le_total
+  decidable_le := String.decidableLE
```
Comment (chabulhwi) [inline Mathlib/Data/String/Basic.lean:242]: This error didn't occur before. Can anyone tell me how to solve this goal?
```
unsolved goals
case mk.mk
⊢ compare { data := data✝¹ } { data := data✝ } = compareOfLessAndEq { data := data✝¹ } { data := data✝ }
```

**Pull request closed without merging**